src/Core/src/Handlers/Layout/LayoutExtensions.cs

```diff
@@ -9,14 +9,34 @@ internal static class LayoutExtensions
 
 		public static int GetLayoutHandlerIndex(this ILayout layout, IView view)
 		{
-			switch (layout.Count)
+			var count = layout.Count;
+			switch (count)
 			{
 				case 0:
 					return -1;
 				case 1:
 					return view == layout[0] ? 0 : -1;
 				default:
-					return layout.OrderByZIndex().IndexOf(view);
+					var found = false;
```

Comment:
A suggestion to improve code readability and optimum performance is to use Span.Sort - with a custom comparer to sort by ZIndex. If one can't sort layout in place one could stackalloc a buffer array to avoid using the heap.

Another idea would be for the layout itself to keep a sorted list of the children.

Reply:
In this case the sort is still less performant than O(N).

A SortedList though would be beneficial for both methods in this class, but the problem is that both zIndex and childIndex might change in time (i.e. adding or removing children).

I'm thinking about a more general solution for the long term, but I felt this was a low hanging fruit for now.

Reply:
It's true that the total thing would be O(N log N) which really isn't much more than O(N) for the low numbers found in layout children. My guess it's probably the allocations of the original code that makes it slower. The sorting itself could in fact be negligible.

Reply:
I do think your code is optimal, I was mainly commenting on readability.